astroid/interpreter/_import/spec.py

```diff
@@ -423,6 +429,18 @@ def _find_spec_with_path(
     raise ImportError(f"No module named {'.'.join(module_parts)}")
 
 
+def spec_cache(func):
+    def wrapper(*args):
+        key = ".".join(args[0])
+        if key not in _spec_cache:
+            _spec_cache[key] = func(*args)
+
+        return _spec_cache[key]
+
+    return wrapper
```

Comment:
Could you add typing to this function?

Reply:
I have typed the function and also the dictionary used for caching (mypy was complaining).